sensing/imu_corrector/src/gyro_bias_estimator.cpp

```diff
@@ -172,36 +225,14 @@ geometry_msgs::msg::Vector3 GyroBiasEstimator::transform_vector3(
 
 void GyroBiasEstimator::update_diagnostics(diagnostic_updater::DiagnosticStatusWrapper & stat)
 {
-  if (gyro_bias_ == std::nullopt) {
-    stat.add("gyro_bias", "Bias estimation is not yet ready because of insufficient data.");
-    stat.summary(diagnostic_msgs::msg::DiagnosticStatus::OK, "Not initialized");
-  } else {
-    stat.add("gyro_bias_x_for_imu_corrector", gyro_bias_.value().x);
-    stat.add("gyro_bias_y_for_imu_corrector", gyro_bias_.value().y);
-    stat.add("gyro_bias_z_for_imu_corrector", gyro_bias_.value().z);
-
-    stat.add("estimated_gyro_bias_x", gyro_bias_.value().x - angular_velocity_offset_x_);
-    stat.add("estimated_gyro_bias_y", gyro_bias_.value().y - angular_velocity_offset_y_);
-    stat.add("estimated_gyro_bias_z", gyro_bias_.value().z - angular_velocity_offset_z_);
-
-    // Validation
-    const bool is_bias_small_enough =
-      std::abs(gyro_bias_.value().x - angular_velocity_offset_x_) < gyro_bias_threshold_ &&
-      std::abs(gyro_bias_.value().y - angular_velocity_offset_y_) < gyro_bias_threshold_ &&
-      std::abs(gyro_bias_.value().z - angular_velocity_offset_z_) < gyro_bias_threshold_;
-
-    // Update diagnostics
-    if (is_bias_small_enough) {
-      stat.add("gyro_bias", "OK");
-      stat.summary(diagnostic_msgs::msg::DiagnosticStatus::OK, "OK");
-    } else {
-      stat.add(
-        "gyro_bias",
-        "Gyro bias may be incorrect. Please calibrate IMU and reflect the result in "
-        "imu_corrector. You may also use the output of gyro_bias_estimator.");
-      stat.summary(diagnostic_msgs::msg::DiagnosticStatus::WARN, "WARN");
-    }
-  }
+  stat.summary(diagnostics_info_.level, diagnostics_info_.summary_message);
+  stat.add("gyro_bias_x_for_imu_corrector", diagnostics_info_.gyro_bias_x_for_imu_corrector);
+  stat.add("gyro_bias_y_for_imu_corrector", diagnostics_info_.gyro_bias_y_for_imu_corrector);
+  stat.add("gyro_bias_z_for_imu_corrector", diagnostics_info_.gyro_bias_z_for_imu_corrector);
+
+  stat.add("estimated_gyro_bias_x", diagnostics_info_.estimated_gyro_bias_x);
+  stat.add("estimated_gyro_bias_y", diagnostics_info_.estimated_gyro_bias_y);
+  stat.add("estimated_gyro_bias_z", diagnostics_info_.estimated_gyro_bias_z);
```

Comment:
To improve visibility when viewing with rqt_runtime_monitor, consider making the following two changes:
* Use `std::fixed`
* Display threshold value

For example, it looks like the following


```cpp
void GyroBiasEstimator::update_diagnostics(diagnostic_updater::DiagnosticStatusWrapper & stat)
{
  auto f = [](const double & value) {
    std::stringstream ss;
    ss << std::fixed << value;
    return ss.str();
  };

  stat.summary(diagnostics_info_.level, diagnostics_info_.summary_message);
  stat.add("gyro_bias_x_for_imu_corrector", f(diagnostics_info_.gyro_bias_x_for_imu_corrector));
  stat.add("gyro_bias_y_for_imu_corrector", f(diagnostics_info_.gyro_bias_y_for_imu_corrector));
  stat.add("gyro_bias_z_for_imu_corrector", f(diagnostics_info_.gyro_bias_z_for_imu_corrector));

  stat.add("estimated_gyro_bias_x", f(diagnostics_info_.estimated_gyro_bias_x));
  stat.add("estimated_gyro_bias_y", f(diagnostics_info_.estimated_gyro_bias_y));
  stat.add("estimated_gyro_bias_z", f(diagnostics_info_.estimated_gyro_bias_z));

  stat.add("gyro_bias_threshold", f(gyro_bias_threshold_));
}
```

When I tried to keep it within the character limit, I ended up with a very short function name :(

If you have any opposing opinions, please let me know.

Reply:
Thank you for the comment!
I followed your recommendation and add the following commit. :+1: